[MP-5763] API: Add Offer and Merchant Product to Shopping List (#827)
spryker/spryker-docs: docs/marketplace/dev/glue-api-guides/202204.0/marketplace-shopping-lists/managing-marketplace-shopping-lists.md

---
title: Managing shopping lists
last_updated: May 18, 2022
template: glue-api-storefront-guide-template
related:
  - title: Glue API - Shopping Lists feature integration
    link: ToDo
  - title: Managing shopping list items
    link: docs/marketplace/dev/glue-api-guides/page.version/shopping-lists/managing-shopping-list-items.html
---

> The Marketplace Shopping Lists API feature allows managing shopping lists in the Marketplace, as well as managing the items in them. 

Reviewer: ```suggestion
The Marketplace Shopping Lists API feature lets you manage shopping lists in the Marketplace, as well as managing the items in them. 
```

| QUERY PARAMETER | DESCRIPTION | EXEMPLARY VALUES |
| --- | --- | --- |

> | include | Adds resource relationships to the request. | shopping-list-items, concrete-products, merchants, product-offers |

Reviewer: Please write the exemplary values as a list. They are better readable in list than in one line

> 

Reviewer: shouldn't we add here a separate message that in order to retrieve merchants for the shopping list items, one should inclulde `shopping list items` and `merchants` resrouces? And to retrieve product offers and their availability in a shopping list, one should include `product-offers` and `product-offer-availabilities`?

| REQUEST SAMPLE | USAGE |
| --- | --- |
| `GET http://glue.mysprykershop.com/shopping-lists/sdb17f85-953f-565a-a4ce-e5cb02405f83` | Retrieve the shopping list with the id `sdb17f85-953f-565a-a4ce-e5cb02405f83`. |
| `GET http://glue.mysprykershop.com/shopping-lists/ecdb5c3b-8bba-5a97-8e7b-c0a5a8f8a74a?include=shopping-list-items` | Retrieve the shopping list with the id `ecdb5c3b-8bba-5a97-8e7b-c0a5a8f8a74a` with its items. |
| `GET http://glue.mysprykershop.com/shopping-lists/ecdb5c3b-8bba-5a97-8e7b-c0a5a8f8a74a?include=shopping-list-items,concrete-products` | Retrieve the shopping list with the id `ecdb5c3b-8bba-5a97-8e7b-c0a5a8f8a74a` with its items and respective concrete products. |

> | `GET http://glue.mysprykershop.com/shopping-lists/c0bc6296-8a0c-50d9-b25e-5bface7671ce?include=shopping-list-items,merchants` | Retrieve the shopping list with the id `c0bc6296-8a0c-50d9-b25e-5bface7671ce` with its merchants. |

Reviewer: ```suggestion
| `GET http://glue.mysprykershop.com/shopping-lists/c0bc6296-8a0c-50d9-b25e-5bface7671ce?include=shopping-list-items,merchants` | Retrieve the shopping list with the id `c0bc6296-8a0c-50d9-b25e-5bface7671ce` with its items and merchants. |
```

> | `GET http://glue.mysprykershop.com/shopping-lists/c0bc6296-8a0c-50d9-b25e-5bface7671ce?include=shopping-list-items,product-offers,product-offer-availabilities`| Retrieve the shopping list with the id `c0bc6296-8a0c-50d9-b25e-5bface7671ce` with its product offers and product offer availabilities. |

Reviewer: ```suggestion
| `GET http://glue.mysprykershop.com/shopping-lists/c0bc6296-8a0c-50d9-b25e-5bface7671ce?include=shopping-list-items,product-offers,product-offer-availabilities`| Retrieve the shopping list with the id `c0bc6296-8a0c-50d9-b25e-5bface7671ce` with its items, product offers, and product offer availabilities. |
```
